malaria24/ona/tests/test_models.py

```diff
@@ -400,16 +402,57 @@ def tearDown(self):
             new_case_alert_case_investigators, sender=ReportedCase)
 
     def get_week(self, cases):
-        start_date = cases \
+        test_cases = ReportedCase.objects.all().order_by("create_date_time")
+        start_date = test_cases \
             .first().create_date_time.strftime(
                 "%d %B %Y"
             )
-        end_date = cases \
+        end_date = test_cases \
             .last().create_date_time.strftime(
                 "%d %B %Y"
             )
         return "{0} to {1}".format(start_date, end_date)
 
+    @responses.activate
+    def test_overall_week_ranges(self):
+        self.mk_actor(role=MANAGER_NATIONAL,
+                      email_address='manager@example.org')
+        ehp1 = self.mk_ehp(name='EHP1', email_address='ehp1@example.org')
+        ehp2 = self.mk_ehp(name='EHP2', email_address='ehp2@example.org')
+        self.mk_mis(name='MIS', email_address='mis@example.org')
+
+        for i in range(10):
+            case = self.mk_case(gender='female', facility_code='342315')
+            case.create_date_time = (datetime(2018, 2,
+                                              random.choice(range(1, 5)),
+                                              11, 30, 30, 0,
+                                              pytz.timezone('US/Pacific'))
+                                     .strftime('%Y-%m-%d %H:%M:%S.%f%z'))
+            case.ehps.add(ehp1)
+            case.ehps.add(ehp2)
```

Comment:
I think for the first loop, we want to add to `ehp1`, and on the second loop, we want to add to `ehp2`, otherwise if we add it to both in both loops, then we're not really checking to see if the problem is solved.

Because if all the ehp's have all of the cases, then the date range for the last ehp is also the date range for all ehps, so we'll get the correct result even if we're just looking at the date range of one of the ehp's.